BASE/Test/Microsoft.ApplicationInsights.Test/Microsoft.ApplicationInsights.Tests/Channel/TransmissionTest.cs

```diff
@@ -501,6 +504,224 @@ public async Task SendAsyncLogsIngestionReponseTimeAndStatusCode()
                     }
                 }
             }
+
+            [TestMethod]
+            public async Task TestTransmissionStatusEventHandlerWithSuccessTransmission()
+            {
+                // ARRANGE
+                var handler = new HandlerForFakeHttpClient
+                {
+                    InnerHandler = new HttpClientHandler(),
+                    OnSendAsync = (req, cancellationToken) =>
+                    {
+                        return Task.FromResult<HttpResponseMessage>(new HttpResponseMessage());
+                    }
+                };
+
+                using (var fakeHttpClient = new HttpClient(handler))
+                {
+                    // Instantiate Transmission with the mock HttpClient                  
+                    Transmission transmission = new Transmission(testUri, new byte[] { 1, 2, 3, 4, 5 }, fakeHttpClient, string.Empty, string.Empty);
+                    
+                    // VALIDATE
+                    transmission.TransmissionStatusEvent += delegate(object sender, TransmissionStatusEventArgs args)
+                    {
+                        Assert.AreEqual(HttpStatusCode.OK, args.Response.StatusCode);
+                    };
+
+                    // ACT
+                    HttpWebResponseWrapper result = await transmission.SendAsync();
+                }
+            }
+
+            [TestMethod]
+            public async Task TestTransmissionStatusEventHandlerWithFailureTransmission()
+            {
+                // ARRANGE
+                var handler = new HandlerForFakeHttpClient
+                {
+                    InnerHandler = new HttpClientHandler(),
+                    OnSendAsync = (req, cancellationToken) =>
+                    {
+                        return Task.FromResult<HttpResponseMessage>(new HttpResponseMessage(HttpStatusCode.ServiceUnavailable));
```

Comment:
both tests are validating scenario when httpclient actually returns a response. We need to test the scenario where HttpClient throws exception. For, eg: timeout, etc.

Reply:
Set timeout to expire immediately and used cancellation token to test it.

Reply:
May I add my 2 cents:
throwing OperationCanceledException instead of returning the HttpResponseMessage from OnSendAsync will also pass the test, but the Assert.AreEqual would not be hit in that case.
Placing Assert outside the delegate would help to solve it.

Reply:
My bad, I didn't see the recent changes. The test will not pass for OperationCanceledException with the new implementation.